### PR TITLE
editoast: detect spacing conflicts

### LINKED ISSUE #4214: editoast: have a conflict detection endpoint
### Description and goal

Now we have resource occupation and reservation timelines, we need to actually perform the conflict detection.

So, actually implement the mock endpoint from #4034

### Acceptance criteria

- [ ] The back-end actually detect and report conflicts through the /timetable/{id}/conflicts endpoint

### PR DESCRIPTION
**Pull request by Khoyo**
Fixes #4214

### CODE REVIEW

File: core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/UndirectedInfraBuilder.java [hunk 1]
```diff
@@ -110,6 +110,8 @@ private void loadCatenaries(List<RJSCatenary> catenaries, HashMap<String, TrackS
 
     private void loadDeadSections(List<RJSDeadSection> deadSections,
                                   HashMap<String, TrackSectionImpl> trackSectionsByID) {
+        if (deadSections == null)
+            deadSections = new ArrayList<>();
```
Comment (multun) [inline core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/UndirectedInfraBuilder.java]: how about just returning?

File: editoast/src/models/train_schedule.rs [hunk 2]
```diff
@@ -156,6 +156,7 @@ pub struct ZoneUpdate {
     pub zone: String,
     pub time: f64,
     pub offset: f64,
+    #[serde(rename = "isEntry")]
```
Comment (multun) [inline editoast/src/models/train_schedule.rs:160]:
```suggestion
    // TODO: see https://github.com/DGEXSolutions/osrd/issues/4294
    #[serde(rename = "isEntry")]
```

File: editoast/src/views/timetable.rs [hunk 3]
```diff
@@ -37,13 +41,17 @@ async fn get(
     Ok(Json(timetable_with_schedules))
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+enum ConflictType {
+    Spacing,
+}
```
Comment (multun) [inline editoast/src/views/timetable.rs:47]:
```suggestion
}

```

File: core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/UndirectedInfraBuilder.java (hunk 1 above)
Comment (Khoyo) [reply core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/UndirectedInfraBuilder.java]: Sure

File: editoast/src/views/timetable.rs [hunk 4]
```diff
@@ -59,25 +67,111 @@ async fn get_conflicts(
...
+                    return Err(
+                        TrainScheduleError::UnsimulatedTrainSchedule { train_schedule_id }.into(),
+                    );
+                }
+                assert!(sim_outputs.len() == 1);
+                Ok(sim_outputs.remove(0))
+            })
+            .await
+            .unwrap()?, // FIXME: maybe this should be converted to an editoast error?
+        );
```
Comment (multun) [inline editoast/src/views/timetable.rs]: this loop runs N sql requests, when N is the number of trains in the timetable; that's not great, can you run a single request instead?

File: editoast/src/views/timetable.rs [hunk 5]
```diff
@@ -59,25 +67,111 @@ async fn get_conflicts(
...
-        train_ids,
-        train_names,
-        start_time: 173,
-        end_time: 1036,
-        conflict_type: "Spacing".to_owned(),
-    };
-    let conflict2 = conflict.clone();
-    Ok(Json(vec![conflict, conflict2]))
+        .map(|conflict| {
+            let start_time = conflict[0].begin.round() as u64; // FIXME: those aren't actually the conflict times
```
Comment (multun) [inline editoast/src/views/timetable.rs]: do you want it merged without the fix?

**Review by multun — CHANGES_REQUESTED** [review]

**Review by multun — APPROVED** [review]

**Pull request merged**